tests/suites/main_test.function

```diff
@@ -237,6 +237,21 @@ int main(int argc, const char *argv[])
 #endif
 #endif
 
+    /* Try changing to the directory containing the executable, if
+     * using the default data file. This allows running the executable
+     * from another directory (e.g. the project root) and still access
+     * the .datax file as well as data files used by test cases
+     * (typically from tests/data_files).
+     *
+     * Note that we do this before the platform setup (which may access
+     * files such as a random seed). We also do this before accessing
+     * test-specific files such as the outcome file, which is arguably
+     * not desirable and should be fixed later.
+     */
+    if (argc == 1) {
```

Comment:
Realise that this is best-effort, but this causes execution from a different directory to fail whenever `--verbose` is passed. We would have to check the nature of the argument to test this though, for example check if it starts with `'-'`.

Reply:
@minosgalanakis just came across this problem

Reply:
Right, we should do the check on the set of list files rather than the commend line arguments, but IIRC that would have required more refactoring between `main_test.function` and `host_test.function` than I cared to do.